### ISSUE TITLE
Changeset comments have individual links, but no button to grab them with

### ISSUE THREAD

**Issue by jidanni** [openstreetmap/openstreetmap-website#2796]
[Here](https://www.openstreetmap.org/changeset/20707962#c596364) I link directly to a changeset comment.

But how did I figure out how?
With Chromium Inspector (F12). That's because I'm a computer wiz.

But for normal mappers, they should be able to get the link from somewhere in the
`Comment from Nuerdensfields 24 minutes ago`
sentence. But due to some bug, they can't!

**Comment by GurminderSingh7** [comment]
i want to help with this issue, could you assign this to me

**Comment by AshBakchod** [comment]
Is this Issue still present or resolved?

**Comment by jidanni** [comment]
Yes, it is still present.

**Comment by pranshu-tomer** [comment]
I can't understand this Issue. Can you give me more clear idea about issue.

**Comment by Dimitar5555** [comment]
> I can't understand this Issue. Can you give me more clear idea about issue.

When you hover over `commented X time ago` there is no link. On other platforms the same text is a link to that specific post/comment. For example in this issue, I can right click on `commented X minutes ago`, copy link and get `https://github.com/openstreetmap/openstreetmap-website/issues/2796#issuecomment-1880081307`.

**Comment by gy-mate** [comment]
As far as I understand the  `.comment_by_html` translations should be changed in all ~90 languages from [`"Comment from %{user} %{time_ago}"`](https://github.com/openstreetmap/openstreetmap-website/blob/29020530a8fabe1d02e3bc9c41970c9030efc380/config/locales/en.yml#L476) to something like `"%{comment} from %{user} %{time_ago}"`¹ in order to be able to send the _Comment_ word as a preformatted link from [here](https://github.com/openstreetmap/openstreetmap-website/blob/29020530a8fabe1d02e3bc9c41970c9030efc380/app/views/changesets/show.html.erb#L35) somehow like this:

```ruby
<%= t comment.visible ? ".comment_by_html" : ".hidden_comment_by_html",
                  :comment => link_to(xxxxxx, "https://www.openstreetmap.org/changeset/#{changeset.id}##{comment.id}"),
                  :time_ago => friendly_date_ago(comment.created_at),
                  :user => link_to(comment.author.display_name, comment.author) %>
```

¹ It's difficult because the word _Comment_ should still be translated in each language, so somehow it should keep the translations _and_ add links to them.

All in all, this seems a bit more complicated to me than a _good first issue_.

**Comment by kleenkanteen** [comment]
@tomhughes Can the "good first issue" label be removed from this because of the above explanation? I got baited into seeing this issue by it.

**Comment by gravitystorm** [comment]
> translations should be changed in all ~90 languages

Yes and no. You don't have to consider how this works, [you only need to make changes to en.yml](https://github.com/openstreetmap/openstreetmap-website/blob/master/CONTRIBUTING.md#i18n) and our translation team will take care of the rest.

> something like "%{comment} from %{user} %{time_ago}"¹

I wouldn't do it quite like that, I would be more likely to make the time_ago into the permalink, just like here on github comments.

> All in all, this seems a bit more complicated to me than a good first issue.

It's a reasonably isolated piece of code, it doesn't require any refactoring or other large-scale changes, so I still think it's reasonable as a good first issue. It might not be a good first issue **for you**, but that's no problem, there's plenty of other things to work on too! :smile:

**Comment by gy-mate** [comment]
> [you only need to make changes to en.yml](https://github.com/openstreetmap/openstreetmap-website/blob/master/CONTRIBUTING.md#i18n) and our translation team will take care of the rest.

My bad!

> I would be more likely to make the time_ago into the permalink, just like here on github comments.

@gravitystorm Ah, so just like this: #4789?

**Comment by gravitystorm** [comment]
> My bad!

No worries, we're here to help you get started and learn how everything works.

> Ah, so just like this: https://github.com/openstreetmap/openstreetmap-website/pull/4789?

More or less! :smile: You'll want to use the path helpers rather than string interpolation into a fixed url, but we can cover this kind of thing in the PR review. Thanks for opening the PR.

**Comment by msudipta888** [comment]
this issue is still open or close because i want to attempt it

**Comment by gy-mate** [comment]
@msudipta888 It's open (as you may see it at the top of the page).
